rc/core/c-family.kak

```diff
@@ -154,7 +154,7 @@ evaluate-commands %sh{
 }
 
 # c specific
-add-highlighter shared/c/code/numbers regex %{\b-?(0x[0-9a-fA-F]+|\d+)[fdiu]?|'((\\.)?|[^'\\])'} 0:value
+add-highlighter shared/c/code/numbers regex %{\b-?(0x[0-9a-fA-F]+|\d+)([fFdDiIuUlL]+)?|'((\\.)?|[^'\\])'} 0:value
```

Comment:
`([fFdDiIuUlL]+)?` == `[fFdDiIuUlL]*`, right ?

Reply:
Yes. This is better. Thanks, I'll fix it

Reply:
I think then we should use `[fFdDiIuUlL]{,3}`

Reply:
because something like `[fFdDiIuU]?[lL]{,2}` requires ordering in suffixes, in terms of the fact, that `ull` is possible, yet `llu` isn't, while it is still valid.

Reply:
I think it's fine to have this kind of false positive, implementing static checks with regex is close to impossible (ever read the regex that validates an email address?). In this particular case, `[fFdDiIuUlL]*` is good enough.

Reply:
with this regexp `([uU][lL]{1,2}|[uU]|[lL]{1,2}|[lL]{1,2}[uU]|[fFdDiI])?` I've managed to make kakoune work similar to vim in term of highlighting suffixes (except i'm not sure about `i` and `d`):
![image](https://user-images.githubusercontent.com/19470159/46292184-ec99bf80-c598-11e8-99fb-b85023818201.png)
Should I use it in this PR?

Reply:
> ever read the regex that validates an email address?

@lenormf Yes........ :cry:

Reply:
I've pushed new regexp. Waiting for your judgement.